app/kumactl/pkg/cmd/root_context.go

```diff
@@ -130,7 +134,68 @@ func (rc *RootContext) CurrentDataplaneTokenClient() (tokens.DataplaneTokenClien
 	if err != nil {
 		return nil, err
 	}
-	return rc.Runtime.NewDataplaneTokenClient(components.Apis.DataplaneToken.LocalUrl)
+	ctx, err := rc.CurrentContext()
+	if err != nil {
+		return nil, err
+	}
+
+	sameMachine, err := rc.cpOnTheSameMachine()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not determine if cp is on the same machine")
+	}
+	var dpTokenUrl string
+	if sameMachine {
+		dpTokenUrl = components.Apis.DataplaneToken.LocalUrl
+	} else {
+		if err := validateRemoteDataplaneTokenServerSettings(ctx, components); err != nil {
+			return nil, err
+		}
+		dpTokenUrl = components.Apis.DataplaneToken.PublicUrl
+	}
+	return rc.Runtime.NewDataplaneTokenClient(dpTokenUrl, ctx.GetCredentials().GetDataplaneTokenApi())
+}
+
+func validateRemoteDataplaneTokenServerSettings(ctx *kumactl_config.Context, components catalogue.Catalogue) error {
+	reason := ""
+	clientConfigured := ctx.GetCredentials().GetDataplaneTokenApi().HasClientCert()
+	serverConfigured := components.Apis.DataplaneToken.PublicUrl == ""
```

Comment:
Should it be `serverConfigured := components.Apis.DataplaneToken.PublicUrl != ""` ?